collectors/modality-probe-collector-common/src/lib.rs

```diff
@@ -283,320 +354,414 @@ impl TryFrom<&LogFileRow> for LogEntry {
-        (arb_probe_id(), any::<String>(), any::<String>()).prop_map(|(id, name, description)| {
-            ProbeMapping {
-                id,
-                name,
-                description,
-            }
-        })
+    pub fn arb_datetime() -> impl Strategy<Value = DateTime<Utc>> {
+        any::<i64>().prop_map(|n| Utc.timestamp_nanos(n))
     }
 
-    pub fn arb_log_entry_data() -> impl Strategy<Value = LogEntryData> {
-        let eid = arb_event_id().prop_map_into().boxed();
-        let lc = (arb_probe_id(), arb_probe_epoch(), arb_probe_clock())
-            .prop_map_into()
-            .boxed();
-        eid.prop_union(lc)
+    pub fn arb_logical_clock() -> impl Strategy<Value = LogicalClock> {
+        (arb_probe_id(), arb_probe_epoch(), arb_probe_clock())
+            .prop_map(|(id, epoch, ticks)| LogicalClock { id, epoch, ticks })
     }
 
-    pub fn arb_datetime() -> impl Strategy<Value = DateTime<Utc>> {
-        any::<i64>().prop_map(|n| Utc.timestamp_nanos(n))
+    pub fn arb_log_entry_data() -> impl Strategy<Value = LogEntryData> {
+        let fc = arb_logical_clock()
+            .prop_map(|lc| LogEntryData::FrontierClock(lc))
+            .boxed();
+        let eid = arb_event_id()
+            .prop_map(|id| LogEntryData::Event(id))
+            .boxed();
+        let eid_wp = (arb_event_id(), any::<u32>())
+            .prop_map(|(id, p)| LogEntryData::EventWithPayload(id, p))
+            .boxed();
+        let tc = arb_logical_clock()
+            .prop_map(|lc| LogEntryData::TraceClock(lc))
+            .boxed();
+        fc.prop_union(eid).or(eid_wp).or(tc)
     }
 
-    pub fn arb_log_entry() -> impl Strategy<Value = LogEntry> {
+    pub fn arb_log_entry() -> impl Strategy<Value = ReportLogEntry> {
         (
             arb_session_id(),
-            arb_segment_id(),
-            arb_segment_index(),
+            arb_sequence_number(),
+            arb_sequence_index(),
             arb_probe_id(),
             arb_log_entry_data(),
             arb_datetime(),
         )
             .prop_map(
-                |(session_id, segment_id, segment_index, probe_id, data, receive_time)| LogEntry {
-                    session_id,
-                    segment_id,
-                    segment_index,
-                    probe_id,
-                    data,
-                    receive_time,
+                |(session_id, sequence_number, sequence_index, probe_id, data, receive_time)| {
+                    ReportLogEntry {
+                        session_id,
+                        sequence_number,
+                        sequence_index,
+                        probe_id,
+                        data,
+                        receive_time,
+                    }
                 },
             )
     }
 
-    prop_compose! {
-        pub(crate) fn gen_raw_user_event_id()(raw_id in 1..=EventId::MAX_USER_ID) -> u32 {
-            raw_id
-        }
-    }
+    #[test]
+    fn report_e2e() {
```

Comment:
Could we please also get a proptest around reporting round-tripping here?  I'm talking about generated Report -> &[u8] -> Report equality checking.

Reply:
:+1: c0adc77eab6c64507ba5d6b58460c064b1aa671d